packages/rfw/test/readme_test.dart

```diff
@@ -0,0 +1,305 @@
+// found in the LICENSE file.
+
+// This file is hand-formatted.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:rfw/formats.dart';
+import 'package:rfw/rfw.dart';
+
+const Map<String, String> rawRemoteWidgetSnippets = <String, String>{
+'root': '''
+// #docregion root
+import local;
+widget root = GreenBox(
+  child: Hello(name: "World"),
+);
+// #enddocregion root
+''',
+
+'fruit': '''
+import local;
+// #docregion fruit
+widget fruit = Foo(
+  bar: { quux: [ 'apple', 'banana', 'cherry' ] },
+);
+// #enddocregion fruit
+''',
+
+'example1': '''
+import local;
+// #docregion example1
+widget example1 = GreenBox(
+  child: Foo(
+    bar: 'Jean',
+  ),
+);
+// #enddocregion example1
+''',
+
+'example2': '''
+import local;
+// #docregion example2
+widget example2 = GreenBox(
+  child: Foo(
+    bar: { name: 'Jean' },
+  ),
+);
+// #enddocregion example2
+''',
+
+'example3': '''
+import local;
+// #docregion example3
+widget example3 = GreenBox(
+  child: Foo(
+    text: ['apple', 'banana']
+  ),
+);
+// #enddocregion example3
+''',
+
+'tap': '''
+import local;
+import core;
+widget tap = GestureDetector(
+  onTap: event 'test' { },
+  child: SizedBox(),
+);
+''',
+
+'tapDown': '''
+import local;
+import core;
+widget tapDown = GestureDetector(
+  onTapDown: event 'test' { },
+  child: SizedBox(),
+);
+''',
+
+'Shop': '''
+// #docregion Shop
+import core;
+
+widget Shop = ListView(
+  children: [
+    Text(text: "Products:"),
+    ...for product in data.server.games:
+      Product(product: product)
+  ],
+);
+
+widget Product = Text(text: args.product.name, softWrap: false, overflow: "fade");
+// #enddocregion Shop
+''',
+
+'MaterialShop': '''
+// #docregion MaterialShop
+import core;
+import material;
+
+widget MaterialShop = Scaffold(
+  appBar: AppBar(
+    title: Text(text: ['Products']),
+  ),
+  body: ListView(
+    children: [
+      ...for product in data.server.games:
+        Product(product: product)
+    ],
+  ),
+);
+
+widget Product = ListTile(
+  title: Text(text: args.product.name),
+  onTap: event 'shop.productSelect' { name: args.product.name, path: args.product.link },
+);
+// #enddocregion MaterialShop
+''',
+};
+
+const String gameData =
+'''
+// #docregion game-data
+{ "games": [
+{"rating": 8.219, "users-rated": 16860, "name": "Twilight Struggle", "rank": 1, "link": "/boardgame/12333/twilight-struggle", "id": 12333},
+{"rating": 8.093, "users-rated": 11750, "name": "Through the Ages: A Story of Civilization", "rank": 2, "link": "/boardgame/25613/through-ages-story-civilization", "id": 25613},
+{"rating": 8.088, "users-rated": 34745, "name": "Agricola", "rank": 3, "link": "/boardgame/31260/agricola", "id": 31260},
+{"rating": 8.082, "users-rated": 8913, "name": "Terra Mystica", "rank": 4, "link": "/boardgame/120677/terra-mystica", "id": 120677},
+// #enddocregion game-data
+// #docregion game-data
+// #enddocregion game-data
```

Comment:
```suggestion
// #enddocregion game-data
```

Or is there something subtle the empty second region is doing?

Reply:
added a comment explaining the subtlety.

Reply:
Ah, I see. The excerpt system sees there's two regions, so it joins them with a `// ...` line, which comes out as a trailing `// ...` line. Clever.